components/Profile/Staking.jsx

```diff
@@ -111,64 +122,135 @@ const StakingDetails = ({ profile }) => {
       serviceInfo?.stakingInstance,
     );
 
+  const handleRestake = async () => {
+    if (!account) return;
+    if (!contractDetails) return;
+    if (!serviceInfo) return;
+
+    setIsRestaking(true);
+
+    try {
+      // Switch to base
+      if (chainId !== base.id) {
+        await switchChainAsync({ chainId: base.id });
+      }
+
+      // First unstake
+      await unstake({ account })
```

Comment:
@kupermind you can review these two calls, restake is unstaking and staking to the same contract as used before